src/Extensions.cs

```diff
@@ -355,8 +360,23 @@ public static void AddImports(this IModelType type, HashSet<string> imports)
         }
 
         public static bool ShouldBeSyntheticType(this IModelType type)
+        {            
+            return (type is PrimaryType || type is SequenceType || type is DictionaryType || type is EnumType || 
+                (type is CompositeTypeGo && (type as CompositeTypeGo).IsPolymorphicResponse()));

```

Comment:
Why do we need to generate a synthetic type for polymorphic responses?

Reply:
Because they need a wrapper to be unmarshalled